src/Tests/Tests/ClientConcepts/HighLevel/Inference/FieldInference.doc.cs

```diff
@@ -563,5 +564,40 @@ public void CodeBasedConfigurationInherits()
 			});
 
 		}
+
+
+		public class SourceModel
+		{
+			[PropertyName("gexo")]
+			public GeoModel Geo { get; set; }
+		}
+
+		public class GeoModel
+		{
+			[JsonProperty("country_iso_code")]
+			public string CountryIsoCode { get; set; }
+		}
+
+		/// this tests runs both during development as wel as during canary builds which reference
+		/// all the artifacts as nuget packages instead of project references
+		[U]
+		public void JsonPropertyIsPickedWhenSerializerIsInternalized()
+		{
+			var usingSettings = WithConnectionSettings(s => s)
+				.WithSourceSerializer(JsonNetSerializer.Default);
+
+			usingSettings.Expect("gexo").ForField(Field<SourceModel>(p=>p.Geo));
+			usingSettings.Expect("country_iso_code").ForField(Field<GeoModel>(p=>p.CountryIsoCode));
+			usingSettings.Expect(new []
+			{
+				"country_iso_code",
+			}).AsPropertiesOf(new GeoModel { CountryIsoCode = "nl" });
+			usingSettings.Expect(new []
+			{
+				"gexo",
+			}).AsPropertiesOf(new SourceModel { Geo = new GeoModel { CountryIsoCode = "nl" } });
+		}
+
+
```

Comment:
I think a test for `Field<SourceModel>(p=>p.Geo.CountryIsoCode)` may also be useful here, for completeness?

Reply:
👍 expanded the unit test to include an assertion for Field inference as well.